### ISSUE TITLE
Creating a TFjob on Kubeflow (API v1) causes Pachd to crash (logs showing 416 HTTP errors)

### ISSUE THREAD

**Issue by Nick-Harvey** [pachyderm/pachyderm#4254]
**What happened?**:
This is probably related to https://github.com/pachyderm/pachyderm/issues/4054. But just in case, here's what happened.

Kubeflow recently released 0.6 and they asked that we update the Pachyderm/Kubeflow example to accommodate the changes before publishing the article. Unfortunately, now when I create the TFjob (by following the same steps before the API change), both the Pachd pod and the TFjob crash shortly after. 

Pachd Logs
```
2019-10-24T15:08:24Z ERROR pfs.BlockAPI.Obj.GetObjects {"duration":0.045474089,"error":"googleapi: got HTTP response code 416 with body: \u003c?xml version='1.0' encoding='UTF-8'?\u003e\u003cError\u003e\u003cCode\u003eInvalidRange\u003c/Code\u003e\u003cMessage\u003eThe requested range cannot be satisfied.\u003c/Message\u003e\u003cDetails\u003ebytes=11490434-\u003c/Details\u003e\u003c/Error\u003e","request":{"objects":[{"hash":"636039fdba2b99748f296645c844e10c449fc392de7b8153b3e050c4d4a4549036529c5005f68b521e8173da6bfb6f6f72be912d97989be702427080ca900c64"}],"offset_bytes":11490434,"total_size":11490434}} []
2019-10-24T15:08:24Z ERROR pfs.API.GetFile {"duration":0.051383112,"error":"rpc error: code = Unknown desc = googleapi: got HTTP response code 416 with body: \u003c?xml version='1.0' encoding='UTF-8'?\u003e\u003cError\u003e\u003cCode\u003eInvalidRange\u003c/Code\u003e\u003cMessage\u003eThe requested range cannot be satisfied.\u003c/Message\u003e\u003cDetails\u003ebytes=11490434-\u003c/Details\u003e\u003c/Error\u003e","request":{"file":{"commit":{"repo":{"name":"input"},"id":"116695cd343a422ca55c625469d2a905"},"path":"mnist.npz"},"offset_bytes":11490434}} []
```

kubectl logs for the pod
```
2019-10-24T15:39:37Z ERROR pfs.BlockAPI.Obj.GetObjects {"duration":0.192019176,"error":"googleapi: got HTTP response code 416 with body: \u003c?xml version='1.0' encoding='UTF-8'?\u003e\u003cError\u003e\u003cCode\u003eInvalidRange\u003c/Code\u003e\u003cMessage\u003eThe requested range cannot be satisfied.\u003c/Message\u003e\u003cDetails\u003ebytes=11490434-\u003c/Details\u003e\u003c/Error\u003e","request":{"objects":[{"hash":"636039fdba2b99748f296645c844e10c449fc392de7b8153b3e050c4d4a4549036529c5005f68b521e8173da6bfb6f6f72be912d97989be702427080ca900c64"}],"offset_bytes":11490434,"total_size":11490434}} []
2019-10-24T15:39:37Z ERROR pfs.API.GetFile {"duration":0.196384095,"error":"rpc error: code = Unknown desc = googleapi: got HTTP response code 416 with body: \u003c?xml version='1.0' encoding='UTF-8'?\u003e\u003cError\u003e\u003cCode\u003eInvalidRange\u003c/Code\u003e\u003cMessage\u003eThe requested range cannot be satisfied.\u003c/Message\u003e\u003cDetails\u003ebytes=11490434-\u003c/Details\u003e\u003c/Error\u003e","request":{"file":{"commit":{"repo":{"name":"input"},"id":"116695cd343a422ca55c625469d2a905"},"path":"mnist.npz"},"offset_bytes":11490434}} []
2019-10-24T15:39:37Z ERROR pfs.BlockAPI.Obj.GetObjects {"duration":0.281795366,"error":"googleapi: got HTTP response code 416 with body: \u003c?xml version='1.0' encoding='UTF-8'?\u003e\u003cError\u003e\u003cCode\u003eInvalidRange\u003c/Code\u003e\u003cMessage\u003eThe requested range cannot be satisfied.\u003c/Message\u003e\u003cDetails\u003ebytes=11490434-\u003c/Details\u003e\u003c/Error\u003e","request":{"objects":[{"hash":"636039fdba2b99748f296645c844e10c449fc392de7b8153b3e050c4d4a4549036529c5005f68b521e8173da6bfb6f6f72be912d97989be702427080ca900c64"}],"offset_bytes":11490434,"total_size":11490434}} []
2019-10-24T15:39:37Z ERROR pfs.API.GetFile {"duration":0.285126825,"error":"rpc error: code = Unknown desc = googleapi: got HTTP response code 416 with body: \u003c?xml version='1.0' encoding='UTF-8'?\u003e\u003cError\u003e\u003cCode\u003eInvalidRange\u003c/Code\u003e\u003cMessage\u003eThe requested range cannot be satisfied.\u003c/Message\u003e\u003cDetails\u003ebytes=11490434-\u003c/Details\u003e\u003c/Error\u003e","request":{"file":{"commit":{"repo":{"name":"input"},"id":"116695cd343a422ca55c625469d2a905"},"path":"mnist.npz"},"offset_bytes":11490434}} []
```

kubectl get pods
```
mysql-ffc889689-w9mqn                                       1/1     Running     0          40m
notebook-controller-deployment-84dbb4c446-vczdg             1/1     Running     0          40m
pachd-6f8b64dddb-nsjpn                                      0/1     OOMKilled   6          17m
profiles-deployment-66d78f5c74-9l5ps                        2/2     Running     0          39m
pytorch-operator-6b687cd959-8p97k                           1/1     Running     0          40m
```

Additional Context:  
1. [Blog article that includes the steps](https://medium.com/@nicksharvey/how-pachyderm-and-kubeflow-make-data-versioning-easy-72e3bc75191a)
2. TFjob page for research [https://www.kubeflow.org/docs/components/training/tftraining/](https://www.kubeflow.org/docs/components/training/tftraining/)
3. [Code and files](https://github.com/pachyderm/pachyderm/tree/master/examples/kubeflow/mnist_with_tfjob)

**What you expected to happen?**:
Job should complete successfully without Pachd crashing. 

**How to reproduce it (as minimally and precisely as possible)?**:
1. Run the [gcp-kubeflow-pachyderm-setup.sh](https://github.com/pachyderm/pachyderm/blob/master/examples/kubeflow/mnist_with_tfjob/gcp-kubeflow-pachyderm-setup.sh) script to get your kubeflow env up and running with Pachyderm installed on GCP.
2. This uses the s3 gateway as part of the tutorial so make sure to register pachyderm.
3. Follow the steps in the blog article to recreate the scenario.
4. Deploy the TFjob and observe

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-20T04:52:26Z", GoVersion:"go1.12.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.10-gke.0", GitCommit:"569511c9540f78a94cc6a41d895c382d0946c11a", GitTreeState:"clean", BuildDate:"2019-08-21T23:28:44Z", GoVersion:"go1.11.13b4", Compiler:"gc", Platform:"linux/amd64"}
```
- Pachyderm CLI and pachd server version (use `pachctl version`): 1.9.7
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): GCP
- OS (e.g. from /etc/os-release): MacOSX High Sierra
- Others:

**Comment by Nick-Harvey** [comment]
Also, Increasing the memory for the Pod does not change the outcome

**Comment by ysimonson** [comment]
I know the 416's were happening because you showed them to me, but I'm not sure what was causing them. Two possibilities that come to mind:

1) The tfjob was repeatedly failing, re-running, and overloading pachd. Still not sure how this should cause 416's, though.
2) An old tfjob was lying around.

Regardless, the issue is no longer reproducible, though there were several bugs with the mnist script. Those issues are fixed with #4482, and running the mnist script there does not trigger any 416's or pachd crashes.
